lgc/builder/BuilderImpl.cpp

```diff
@@ -658,7 +657,61 @@ static bool instructionsEqual(Instruction *lhs, Instruction *rhs) {
 
   return true;
 }
-#endif
+
+// =====================================================================================================================
+// Exttract the 32-bit value of the non-uniform index.
+// @param nonUniformIndex : the non-uniform index of the non-uniform operand of the image call
+// @return : the 32-bit value of the nonUniformIndex
+Value *get32BitNonUniformIndex(Value *nonUniformIndex) {
+  if (nonUniformIndex->getType()->isIntegerTy(64)) {
```

Comment:
The only invocation of `get32BitNonUniformIndex` is guarded with the same check `nonUniformIndex->getType()->isIntegerTy(64)`. Either remove it there or keep it and add an assert here.